File: Assembly-FSharp/Assembly-FSharp/Commands/InputKeyDownCommand.fs
namespace Styx.Commands

open UnityEngine

type InputKeyDownCommand() =
    inherit Command ()

    [<SerializeField>]
    let mutable inputKey: string = ""
    [<SerializeField>]
    let mutable isPressed: bool = false

    member this.IsPressed = isPressed

    override this.Execute () =
        isPressed <- (Input.GetButtonDown inputKey)

File: Assets/_Scripts/Commands/InputKeyDownCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;

namespace Styx.Commands
{
    public class InputKeyDownCommand : Command
    {
        [SerializeField]
        private string _inputKey = "";
        [SerializeField]
        private bool _isPressed = false;

        public bool IsPressed
        {
            get { return _isPressed; }
        }

        public override void Execute()
        {
            _isPressed = Input.GetButtonDown(_inputKey);
        }
    }
}
